docs/guide/migration.md

```diff
@@ -1,73 +1,46 @@
-# Migration from v3
+# Migration from v4
 
-## Rollup 3
+## Node.js Support
 
-Vite is now using [Rollup 3](https://github.com/vitejs/vite/issues/9870), which allowed us to simplify Vite's internal asset handling and has many improvements. See the [Rollup 3 release notes here](https://github.com/rollup/rollup/releases/tag/v3.0.0).
+Vite no longer supports Node.js 14 / 16 / 17 / 19, which reached its EOL. Node.js 18 / 20+ is now required.
 
-Rollup 3 is mostly compatible with Rollup 2. If you are using custom [`rollupOptions`](../config/build-options.md#rollup-options) in your project and encounter issues, refer to the [Rollup migration guide](https://rollupjs.org/migration/) to upgrade your config.
+## Deprecate CJS node API
 
-## Modern Browser Baseline change
-
-The modern browser build now targets `safari14` by default for wider ES2020 compatibility (bumped from `safari13`). This means that modern builds can now use [`BigInt`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt) and that the [nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) isn't transpiled anymore. If you need to support older browsers, you can add [`@vitejs/plugin-legacy`](https://github.com/vitejs/vite/tree/main/packages/plugin-legacy) as usual.
+TODO: add some description https://github.com/vitejs/vite/pull/14278
 
 ## General Changes
 
-### Encoding
-
-The build default charset is now utf8 (see [#10753](https://github.com/vitejs/vite/issues/10753) for details).
-
-### Importing CSS as a String
-
-In Vite 3, importing the default export of a `.css` file could introduce a double loading of CSS.
-
-```ts
-import cssString from './global.css'
-```
-
-This double loading could occur since a `.css` file will be emitted and it's likely that the CSS string will also be used by the application code — for example, injected by the framework runtime. From Vite 4, the `.css` default export [has been deprecated](https://github.com/vitejs/vite/issues/11094). The `?inline` query suffix modifier needs to be used in this case, as that doesn't emit the imported `.css` styles.
-
-```ts
-import stuff from './global.css?inline'
-```
-
-### Production Builds by Default
+### Allow path containing `.` to fallback to index.html
 
-`vite build` will now always build for production regardless of the `--mode` passed. Previously, changing `mode` to other than `production` would result in a development build. If you wish to still build for development, you can set `NODE_ENV=development` in the `.env.{mode}` file.
+In Vite 4, accessing a path containing `.` did not fallback to index.html even if `appType` is set to `'SPA'` (default).
+From Vite 5, it will fallback to index.html.
```

Comment:
@btea Do you have any suggestions about how to explain it?
https://github.com/vitejs/vite/pull/14142#issuecomment-1710284252

Reply:
I'm not sure how to describe it, but it's better to write into the documentation the 404 error mentioned in the previous comment that the console won't throw if you can't find the picture.

Reply:
I put a message about it.